x-pack/legacy/plugins/apm/public/context/ChartsTimeContext.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, useState } from 'react';
+
+const ChartsTimeContext = React.createContext<{
+  time: number | null;
+  setTime: (time: number | null) => unknown;
+} | null>(null);
+
+const ChartsTimeContextProvider: React.FC = ({ children }) => {
```

Comment:
Simply syncs time across all charts in its context.

Reply:
This sounds similar to
https://github.com/elastic/kibana/blob/7056f6143d7863c2b48ff22ee376402c932bbb60/x-pack/legacy/plugins/apm/public/components/shared/charts/SyncChartGroup/index.tsx